shelley/chain-and-ledger/executable-spec/src/BlockChain.hs

```diff
@@ -207,54 +214,50 @@ slotsPrior = 33 -- one third of slots per epoch
 startRewards :: Duration
 startRewards = 33 -- see above
 
-verifyVrf
-  :: (DSIGNAlgorithm dsignAlgo, Eq a)
-  => VKey dsignAlgo
-  -> Seed
-  -> (a, Proof dsignAlgo a)
-  -> Bool
-verifyVrf vk seed (val, Proof k s v) = vk == k && seed == s && val == v
-
 vrfChecks
-  :: (HashAlgorithm hashAlgo, DSIGNAlgorithm dsignAlgo)
-  => Seed
-  -> PoolDistr hashAlgo dsignAlgo
+  ::  ( HashAlgorithm hashAlgo
+      , DSIGNAlgorithm dsignAlgo
+      , VRFAlgorithm vrfAlgo
+      , VRF.Signable vrfAlgo Nonce
+      )
+  => Nonce
+  -> PoolDistr hashAlgo dsignAlgo vrfAlgo
   -> UnitInterval
-  -> BHBody hashAlgo dsignAlgo kesAlgo
+  -> BHBody hashAlgo dsignAlgo kesAlgo vrfAlgo
   -> Bool
 vrfChecks eta0 (PoolDistr pd) f bhb =
   let sigma' = Map.lookup hk pd
   in  case sigma' of
         Nothing -> False
         Just (sigma, vrfHK) ->
-          vrfHK == hashKey vrfK
-            && verifyVrf vrfK
-                         ((eta0 ⭒ ss) ⭒ SeedEta)
-                         (bheaderEta bhb, bheaderPrfEta bhb)
-            && verifyVrf vrfK
-                         ((eta0 ⭒ ss) ⭒ SeedL)
-                         (bheaderL bhb, bheaderPrfL bhb)
-            && intervalValue (bheaderL bhb)
+          vrfHK == hashKeyVRF vrfK
+            && VRF.verifyCertified vrfK
+                         ((eta0 ⭒ ss) ⭒ seedEta)
```

Comment:
I think `seedEta` and `seedL` are supposed to be handled differently than the other nonces.  Instead of using the seed operation (which is now correctly being implemented as a hash, thanks!), these should be `XOR`ed.